### PR TITLE
test: check for 2 ASSIGN for >=8.7.0 & snapshot

### PR DESCRIPTION
**Pull request by matthewBearCamunda**
## Description

See https://camunda.slack.com/archives/C0622NWC9LK/p1733769293022809 for context

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

**Comment by matthewBearCamunda** [comment]
I can run the zeebe compatibility job manually

See https://github.com/camunda/camunda/actions/runs/12259915258 as an example run (this is the first one)

File: optimize/backend/src/it/java/io/camunda/optimize/AbstractCCSMIT.java [hunk 1]
```diff
@@ -338,11 +362,34 @@ protected OffsetDateTime getTimestampForZeebeAssignEvents(
         Instant.ofEpochMilli(startOfElement.getTimestamp()), ZoneId.systemDefault());
   }
 
+  protected OffsetDateTime getTimestampForZeebeLastAssignEvents(
```
Comment (RomanJRW) [inline optimize/backend/src/it/java/io/camunda/optimize/AbstractCCSMIT.java:365]:
```suggestion
  protected OffsetDateTime getTimestampForZeebeLastAssignedEvents(
```
Just for correctness to align it with the intent

File: optimize/backend/src/it/java/io/camunda/optimize/service/importing/ZeebeUserTaskImportIT.java [hunk 2]
```diff
@@ -917,6 +950,11 @@ private OffsetDateTime getTimestampForAssignedUserTaskEvents(
     return getTimestampForFirstZeebeEventsWithIntent(eventsForElement, ASSIGNED);
   }
 
+  private OffsetDateTime getTimestampForAssignedLastUserTaskEvents(
+      final List<ZeebeUserTaskRecordDto> eventsForElement) {
+    return getTimestampForLastZeebeEventsWithIntent(eventsForElement, ASSIGNED);
+  }
```
Comment (RomanJRW) [inline optimize/backend/src/it/java/io/camunda/optimize/service/importing/ZeebeUserTaskImportIT.java:956]: NABD but not sure this helper adds much value, as the proxied method already has reasonably good naming and interface

File: optimize/backend/src/it/java/io/camunda/optimize/service/importing/ZeebeUserTaskImportIT.java [hunk 3]
```diff
@@ -938,6 +976,14 @@ private long getDurationInMsBetweenStartAndFirstAssignOperation(
         .toMillis();
   }
 
+  private long getDurationInMsBetweenStartAndLastAssignOperation(
```
Comment (RomanJRW) [inline optimize/backend/src/it/java/io/camunda/optimize/service/importing/ZeebeUserTaskImportIT.java:979]:
```suggestion
  private long getDurationInMsBetweenStartDateAndLastAssignedOperation(
```
Just to clarify that the duration is not between different assigned events, but between the actual start date and the last assigned event

File: optimize/backend/src/it/java/io/camunda/optimize/AbstractCCSMIT.java [hunk 4]
```diff
@@ -338,11 +362,34 @@ protected OffsetDateTime getTimestampForZeebeAssignEvents(
...
+
   protected OffsetDateTime getTimestampForZeebeUnassignEvent(
       final List<? extends ZeebeRecordDto> eventsForElement) {
     return getTimestampForZeebeAssignEvents(eventsForElement, "");
   }
 
+  protected OffsetDateTime getTimestampForZeebeLastUnassignEvent(
+      final List<? extends ZeebeRecordDto> eventsForElement) {
+    return getTimestampForZeebeLastAssignEvents(eventsForElement, "");
+  }
```
Comment (RomanJRW) [inline optimize/backend/src/it/java/io/camunda/optimize/AbstractCCSMIT.java:391]: Also not sure that this wrapper method is super helpful, as the called method is quite indicative of behaviour already

File: optimize/backend/src/it/java/io/camunda/optimize/service/importing/ZeebeUserTaskImportIT.java [hunk 5]
```diff
@@ -546,9 +546,22 @@ public void importUnclaimOperation_viaWriter() {
         deployAndStartInstanceForProcess(
             createSimpleNativeUserTaskProcessWithAssignee(TEST_PROCESS, null, ASSIGNEE_ID));
     waitUntilUserTaskRecordWithElementIdExported(USER_TASK);
-    zeebeExtension.unassignUserTask(
-        getExpectedUserTaskInstanceIdFromRecords(getZeebeExportedUserTaskEvents()));
-    waitUntilUserTaskRecordWithIntentExported(ASSIGNED);
+
+    final boolean isZeebeVersionSnapshotForHigherThan870 =
+        isZeebeVersion87OrLater() || isZeebeVersionSnapshot();
+    if (isZeebeVersion87OrLater() || isZeebeVersionSnapshot()) {
```
Comment (RomanJRW) [inline optimize/backend/src/it/java/io/camunda/optimize/service/importing/ZeebeUserTaskImportIT.java:552]:
```suggestion
    if (isZeebeVersionSnapshotForHigherThan870) {
```

File: optimize/backend/src/it/java/io/camunda/optimize/service/importing/ZeebeUserTaskImportIT.java [hunk 6]
```diff
@@ -546,9 +546,22 @@ public void importUnclaimOperation_viaWriter() {
         deployAndStartInstanceForProcess(
             createSimpleNativeUserTaskProcessWithAssignee(TEST_PROCESS, null, ASSIGNEE_ID));
     waitUntilUserTaskRecordWithElementIdExported(USER_TASK);
-    zeebeExtension.unassignUserTask(
-        getExpectedUserTaskInstanceIdFromRecords(getZeebeExportedUserTaskEvents()));
-    waitUntilUserTaskRecordWithIntentExported(ASSIGNED);
+
+    final boolean isZeebeVersionSnapshotForHigherThan870 =
```
Comment (RomanJRW) [inline optimize/backend/src/it/java/io/camunda/optimize/service/importing/ZeebeUserTaskImportIT.java:550]:
```suggestion
    final boolean isZeebeVersionHigherThan870 =
```
I think SNAPSHOT is already implicitly included in the "higher" part of this description

File: optimize/backend/src/it/java/io/camunda/optimize/service/importing/ZeebeUserTaskImportIT.java [hunk 7]
```diff
@@ -610,8 +625,21 @@ public void importUnclaimOperation_viaUpdateScript() {
     importAllZeebeEntitiesFromScratch();
 
     List<ZeebeUserTaskRecordDto> exportedEvents = getZeebeExportedUserTaskEvents();
-    zeebeExtension.unassignUserTask(getExpectedUserTaskInstanceIdFromRecords(exportedEvents));
-    waitUntilUserTaskRecordWithIntentExported(ASSIGNED);
+
+    final boolean isZeebeVersionSnapshotForHigherThan870 =
```
Comment (RomanJRW) [inline optimize/backend/src/it/java/io/camunda/optimize/service/importing/ZeebeUserTaskImportIT.java:629]:
```suggestion
    final boolean isZeebeVersionHigherThan870 =
```
As above

**Review by RomanJRW — APPROVED** [review]
Nice managing of the conditionals without having to write a new test! Pre-approve, with mostly naming suggestions 👍

**Pull request merged**